### DISCUSSION TITLE
Can we use the "SiteMigration" template for all installations?

### DISCUSSION THREAD

**Discussion by wcrisafu** [oqtane/oqtane.framework#2738] [Q&A] [answered]
This may sound like a strange request but I am hoping there is a way to automate module installation and configuration. I would like roles, pages (with specific modules) automatically created and configured when I install my module to a new site/database. How do i modify "[SiteMigration("localhost:44357", "01.00.01")]" to make it generic run for all installs regardless of URL or port?

regards,

**Comment by leigh-pointer** [comment]
Add your permission in the moduleinfo class

**Comment by sbwalker** [comment, marked as answer]
Currently a site migration is related to a specific alias - however it would be possible to enhance the framework to allow for a wildcard (ie. "*") specification.

**Reply by wcrisafu** [reply]
Thank you. I think it would be a nice feature to add to a package to install with pre-configured pages/roles/settings.

**Comment by sbwalker** [comment]
@leigh-pointer this item is related to the Site Migration feature introduced in Oqtane 3.1: https://www.oqtane.org/blog/!/38/site-migrations-deployment

**Comment by sbwalker** [comment]
@wcrisafu After reading your requirements more closely, I do not believe a SiteMigration is the correct approach. If you want to create pages, etc... when you install/upgrade your module you should use the IInstallable interface which is declared in the module's Manager class:

```
public bool Install(Tenant tenant, string version)
{
    // include your custom logic here

    return Migrate(new YourModuleContext(_DBContextDependencies), tenant, MigrationType.Up);
}
```

This allows you to execute versioned logic which is specific to your module. This will be executed during Startup, the same way that SiteMigrations are executed.
